docs/workflow/developer-workflows.md

```diff
@@ -0,0 +1,441 @@
+
+## Helm Workflows and User Experience
+
+In this section we examine several workflows that we feel are central to
+the experience of deploying, managing, and building applications using
+Helm. Each workflow is followed by a basic model of where the processing
+of commands occurs.
+
+### User Workflow
+
+The user workflow is the common case. This answers stories for the "tire
+kicker" and "standard user" personas.
+
+#### Simple deployment:
+
+```
+$ helm deploy helm:example.com/foo/bar
+Created wonky-panda
+```
+
+- The client sends the server a request to deploy `helm:example.com/foo/bar`.
+- The server assigns a random name `wonky-panda`, fetches the chart from
+  object storage, and goes about the deployment process.
+
+#### Find out about params:
+
+In this operation, helm reads a chart and returns the list of parameters
+that can be supplied in a template:
+
+```
+$ helm list-params helm:example.com/foo/bar
+Params:
+- bgcolor: The background color for the home page (hex code or HTML colors)
+- title: The title of the app
+```
+
+- The client sends the request to the API server
+- The API server fetches the chart, analyzes it, and returns the list of
+  parameters.
+
+#### Generate the params for me:
+
+In this operation, helm generates a parameter values file for the user.
+
+```
+$ helm gen-params helm:example.com/foo/bar
+Created: values.yaml
+$ edit values.yaml
+```
+
+- The client sends the request to the server
+- The server returns a stub file
+- The client writes the file to disk
+
+#### Deploy with params:
+
+In this operation, the user deploys a chart with an associated values
+file.
+
+```
+$ helm deploy helm:example.com/foo/bar values.yaml
+Created taco-tuesday
+```
+
+- The client sends a request with the name of the chart and the values
+  file.
+- The API server generates a name (`taco-tuesday`)
+- The API server fetches the request chart, sends the chart and values
+  through the template resolution phase, and deploys the results.
+
+If we allow the user to pass in a name (overriding the generated name),
+then the server must first guarantee that this name is unique.
+
+#### Get the info about named deployment.
+
+A deployment, as we have seen, is a named instance of a chart.
+Operations that operate on these instances use the name to refer to the
+instance.
+
+```
+$ helm status taco-tuesday
+OK
+Located at: 10.10.10.77:8080
+```
+
+- The client sends the API server a request for the status of
+  `taco-tuesday`.
+- The API server looks up pertinent data and returns it.
+- The client displays the data.
+
+#### Edit and redeploy:
+
+Redeployment is taking an existing _instance_ and changing its template
+values, and then re-deploying it.
+
+```
+$ edit values.yaml
+$ helm redeploy taco-tuesday values.yaml
+Redeployed taco-tuesday
+```
+
+- The client sends the instance name and the new values to the server
+- The server coalesces the values into the existing instance and then
+  restarts.
+
+#### Uninstall:
```

Comment:
Uninstall seems odd, since we didn't install taco-tuesday. Delete might be better.